druid/src/tests/mod.rs

```diff
@@ -14,18 +14,43 @@
 
 //! Additional unit tests that cross file or module boundaries.
 
-mod harness;
+pub(crate) mod harness;
 mod helpers;
 mod layout_tests;
 
 use std::cell::Cell;
+use std::env;
+use std::fs;
 use std::rc::Rc;
+use tempfile;
 
 use crate::widget::*;
 use crate::*;
 use harness::*;
 use helpers::*;
 
+/// This function creates a temperary direcotry and returns a Path buf to it.
+///
+/// This directory will be created relative to the exicutable and will there for
+/// be created in the target directory for tests run with cargo. The directory will
+/// be cleaned up at the end of the Path bufs lifetime. This
+/// uses the `tempfile` crate.
```

Comment:
*~temperary~ temporary*
*~Path buf~ \`PathBuf\`* (2x)
*~exicutable~ executable*
*~there for~ therefore*
*~run with cargo~ when running with cargo*